modules/generic-simple/src/main/scala/io/circe/generic/simple/codec/DerivedAsObjectCodec.scala

```diff
@@ -0,0 +1,23 @@
+package io.circe.generic.simple.codec
+
+import io.circe.{ Codec, Decoder, HCursor, JsonObject }
+import shapeless.LabelledGeneric
+
+abstract class DerivedAsObjectCodec[A] extends Codec.AsObject[A]
+
+final object DerivedAsObjectCodec {
+  implicit def deriveCodec[A, R](
+    implicit
+    gen: LabelledGeneric.Aux[A, R],
+    codec: => ReprAsObjectCodec[R]
+  ): DerivedAsObjectCodec[A] = new DerivedAsObjectCodec[A] {
+    final def apply(c: HCursor): Decoder.Result[A] = codec.apply(c) match {
```

Comment:
Do you not need to cache code in a lazy val to avoid executing it again and again for each call to apply?

Reply:
Yeah, I meant to go back and do a pass for runtime performance stuff like this after getting it correct. I've just opened #1183. Thanks for catching.